File: LL.java
    public class LL {
    class Node{
        int data;
        Node next;

        Node(int data){
            this.data = data;
            this.next = null;
        }
    }
    Node first = null;

    public void insertAtFirst(int data){
        Node newNode = new Node(data);
        if(first == null){
            first = newNode;
            return;
        }

        newNode.next = first;
        first = newNode;
    }

    public void insertAtLast(int data){
        Node newNode = new Node(data);

        if(first == null){
            first = newNode;
            return;
        }

        Node currentNode = first;
        while(currentNode.next != null){
            currentNode = currentNode.next;
        }
        currentNode.next = newNode;
    }


    public void deleteFirst(){
        if(first == null){
            System.out.println(" ");
            System.out.println("Linked List is Empty!!");
            return;
        }
        first = first.next;
    }

    public void deleteLast(){
        if(first == null){
            System.out.println(" ");
            System.out.println("Linked List is Empty!!");
        }

        if(first.next == null){
            first = null;
            return;
        }
        Node secondLast = first;
        Node last = first.next;

        while(last.next != null){
            secondLast = secondLast.next;
            last = last.next;
        }
        secondLast.next = null;
    }

    public void insertOrd(int x){
        Node newNode = new Node(x);

        if(first == null || newNode.data <= first.data){
            newNode.next = first;
            first = newNode;
            return;
        }

        Node currentNode = first;

        while(currentNode.next != null && newNode.data >= currentNode.next.data){
            currentNode = currentNode.next;
        }
        newNode.next = currentNode.next;
        currentNode.next = newNode;
    }

    public void deleteFromElmt(int x){
        Node currNode = first;

        if(first == null){
            System.out.println(" ");
            System.out.println("Linked List Is Empty!!");
            return;
        }

        if(currNode.data == x){
            first = currNode.next;
            return;
        }

        while(currNode.next.data!=x && currNode.next.next != null){
            currNode = currNode.next; 
        }

        currNode.next = currNode.next.next;
    }
    
    public void printList(){

        if(first == null){
            System.out.println("LinkedList is Empty");
            return;
        }
        Node cuurrentNode = first;
        
        while(cuurrentNode != null){
            System.out.print(cuurrentNode.data + " -> ");
            cuurrentNode = cuurrentNode.next;
        }
        System.out.print("null");
        System.out.println(" ");
    }

    public static void main(String[] args) {

        LL list = new LL();
    
        list.insertAtFirst(1);
        list.insertAtLast(2);
        list.insertAtLast(7);
        list.insertAtLast(8);

        list.insertOrd(3);
        list.printList();
    }
}

File: LL.c
#include<stdio.h>
#include<stdlib.h>

typedef struct Node
{
    int data;
    struct Node* next;
}Node;

Node* createNode(int data){

        Node* newNode = (Node*)(malloc(sizeof(Node)));

        newNode->data = data;
        newNode->next = 0;

        return newNode;
}

Node* insertAtFirst(Node* first,int val){

    Node* newNode = createNode(val);

    if(first == NULL){
        return newNode;
    }

    newNode -> next = first;
    return newNode;
}

Node* insertAtLast(Node* first,int val){

    Node* newNode = createNode(val);
    if(first == NULL){
        return newNode;
    }

    Node* curr = first;

    while(curr -> next){
        curr = curr -> next;
    }

    curr -> next = newNode;
    return first;
}

Node* deleteFirst(Node* first){
    if(first == NULL){
        printf("LinkedList is empty.");
    }
    first = first -> next;
    return first;
}

Node* deleteLast(Node* first){

    if(first == NULL){
        printf("LinkedList in empty.");
    }

    if(first -> next == NULL) first = NULL;

    Node* curr = first;

    while(curr -> next -> next){
        curr = curr -> next;
    }

    curr -> next = NULL;

    return first;
}

void printList(Node* first){

    if(first == NULL){
        printf("List is Empty.");
        return;
    }

    Node* curr = first;

    while(curr){
        printf("%d -> ",curr -> data);
        curr = curr -> next;
    }

    printf("null");
}

int main(){

    Node* first = NULL;
    first = createNode(1);

    first = insertAtFirst(first,2);
    first = insertAtFirst(first,3);
    first = insertAtLast(first,10);

    first = deleteFirst(first);
    first = deleteLast(first);

    printList(first);

    return 0;

}